Data/ByteString/Streaming/Char8.hs

```diff
@@ -713,8 +695,7 @@ putStrLn bs = hPut IO.stdout (snoc bs '\n')
 -- , count
 -- , count'
 
-{-| This will read positive or negative Ints that require 18 or fewer characters.
--}
+-- | This will read positive or negative Ints that require 18 or fewer characters.
```

Comment:
The choice of `18` here is rather puzzling.  The 64-bit `minBound` (`-9223372036854775808`) takes 20 bytes, and `maxBound` (`9223372036854775807`) takes 19 bytes, so whence 18???

Reply:
A more robust function might read up to 21 bytes, parse the input as an `Integer` and fail (i.e. return Nothing) if the result is too big to be an `Int`.  This version silently truncates to 32-bits on 32-bit systems (increasingly rare), but also silently truncates integers that require 19 or more digits, leaving tail digits in the stream...

Reply:
Good question, I can't say for sure.

Reply:
Don't know whether you subscribe to the haskell-cafe list, but in a post there today, I mentioned in passing that the `readInt` in this module is also egregiously slow. I got a 5x performance improvement when I wrote my own, and it entirely avoids the bogus 18-char limit, which is semantically dubious (wrong!) and cause overly long strings of decimals to parsed as multiple ints in successive calls to `readInt` for example.

So (likely a separate PR), we should replace `readInt` with something like my (not yet 100% polished) version...